src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkCorrectUsingMethodsInspection.kt

```diff
@@ -23,24 +24,20 @@ class SmkCorrectUsingMethodsInspection: SnakemakeInspection() {
             session: LocalInspectionToolSession
     ) = object : SnakemakeInspectionVisitor(holder, session) {
 
-        private fun checkAncient(psiEl: PsiElement): PsiElement? {
-            if (psiEl.text != SECTION_INPUT)
-                return psiEl
-            return null
-        }
+        private fun getAllowedSectionNames(ruleOrCheckpointSectionName: String): ArrayList<String>? {
 
-        private fun checkProtected(psiEl: PsiElement): PsiElement? {
-            val psiElText = psiEl.text
-
-            if (psiElText != SECTION_OUTPUT
-                    && psiElText != SECTION_LOG
-                    && psiElText != SECTION_BENCHMARK)
-                return psiEl
-            return null
+            return when (ruleOrCheckpointSectionName) {
+                METHOD_ANCIENT -> arrayListOf(SECTION_INPUT)
+                METHOD_PROTECTED -> arrayListOf(SECTION_OUTPUT, SECTION_LOG, SECTION_BENCHMARK)
+                METHOD_DIRECTORY, METHOD_TEMP -> arrayListOf(SECTION_OUTPUT)
```

Comment:
>> Please check other methods from same source file, e.g. temp(..) and etc

> I added temp and report methods, I'm not sure about their usage, should be check


@ArtemDavletov You are the author of this feature so it's your responsibility to do the best and to find all related methods and check their usage in documentation/source code/at runtime. I'm ready to discuss the details/impl/etc and any troubles during such investigation, but please do not mark this PR as `done` if you are not sure what you are doing and how it works. If author doesn't understand whether his feature works or not - it is quite bad for the project. I'm sure that you can figure out what possible flags could be used here and how they are supposed to be used.